google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -110,16 +111,22 @@ JLabel getWarningMessage() {
   }
 
   public boolean isModified() {
-    return !Comparing.strEqual(getCloudSdkDirectory(),
-        CloudSdkService.getInstance().getCloudSdkHomePath());
+    if (getCloudSdkDirectory() != null) {
+      return Paths.get(getCloudSdkDirectory())
+          .compareTo(CloudSdkService.getInstance().getCloudSdkHomePath()) != 0;
+    }
+
+    return CloudSdkService.getInstance().getCloudSdkHomePath() != null;
```

Comment:
Moving my comment to the right place:

What about the case where one is null, and the other is empty? Just trying to make sure this does what we expect: e.g. if the field is empty string, but the SDK from the service is null, will this return true?